R/ml_persistence.R

```diff
@@ -85,15 +85,45 @@ ml_save.ml_model <- function(x, path, overwrite = FALSE,
 #' @rdname ml-persistence
 #' @export
 ml_load <- function(sc, path) {
-  is_local <- spark_context(sc) %>%
-    invoke("isLocal")
-
+  is_local <- spark_context(sc) %>% invoke("isLocal")
+  path <- cast_string(path)
   if (is_local) {
-    path <- cast_string(path) %>%
-      spark_normalize_path()
+    path <- spark_normalize_path(path)
   }
 
-  metadata <- try(
+  metadata_glob <- file.path(path, "metadata", "part-*")
+  metadata_view <- random_string("ml_load_metadata")
+
+  df_try <- try(
+    spark_read_json(
+      sc   = sc,
+      name = metadata_view,
+      path = metadata_glob
+    ),
+    silent = TRUE
+  )
+
+  if (!inherits(df_try, "try-error")) {
+    class_val <- NULL
+
+    cls1 <- try(dplyr::pull(df_try, "class"), silent = TRUE)
```

Comment:
I think if we can change this `try()` call with  `df_try_cols <- colnames(df_try)` and then interrogate here, and on line 114 simply check if `df_try_cols` contains those columns. Is there a specific reason to use `try()`?